core/translation.cpp

```diff
@@ -848,45 +957,143 @@ void Translation::set_locale(const String &p_locale) {
 	}
 }
 
-void Translation::add_message(const StringName &p_src_text, const StringName &p_xlated_text) {
-	translation_map[p_src_text] = p_xlated_text;
+void Translation::set_plural_rule(const String &p_plural_rule) {
+	// Set plural_forms and plural_rule.
+	// p_plural_rule passed in has the form "Plural-Forms: nplurals=2; plural=(n >= 2);".
```

Comment:
```suggestion
void Translation::set_plural_rule(const String &p_plural_rule) {
	// Set plural_forms and plural_rule.
	// p_plural_rule passed in has the form "Plural-Forms: nplurals=2; plural=(n >= 2);".
```

I'm a bit concerned about having the gettext-specific plural handling directly in the base `Translation`, which is also used for CSV and could theoretically be used by more input formats in the future (though we have no plans to support more than PO and CSV).

Maybe this should be abstracted some more and the gettext-specific logic should be in a `TranslationPO` Resource?

WDYT @vnen @YeldhamDev?